### PR TITLE
Fix session_regenerate_id

### PR DESCRIPTION
**Pull request by lukyer**
Fixes `session_regenerate_id(): Session object destruction failed` which sometimes occures.

See same fix: https://github.com/yiisoft/yii/issues/2376

### CODE REVIEW

File: src/Http/Session.php [hunk 1]
```diff
@@ -226,7 +226,7 @@ public function regenerateId()
 			if (headers_sent($file, $line)) {
 				throw new Nette\InvalidStateException('Cannot regenerate session ID after HTTP headers have been sent' . ($file ? " (output started at $file:$line)." : '.'));
 			}
-			session_regenerate_id(TRUE);
+			if (session_id() !== '') session_regenerate_id(TRUE);
```
Comment (enumag) [inline src/Http/Session.php]: Use multi-line if with `{ ... }`.

**Comment by dg** [comment]
Can you squash it to single commit and use consistent commit message?

**Comment by lukyer** [comment]
I really tried to, but you can see the mess i made using tortoisegit.
What tool do you use for rebase?

**Comment by milo** [comment]
Use command line. I suppose, you have your Github fork as an `origin` remote.
```
cd http
git remote add upstream https://github.com/nette/http.git
git fetch upstream
git checkout patch-1
git rebase upstream/master
git rebase -i HEAD~20  # or how many commits you want to manage/squash/rename
# rewrite all 'pick' to 's' (as squash) and save
git push origin patch-1 --force
```

**Comment by lukyer** [comment]
@milo thank you for your advice, all my complications were caused by missing `--force` flag during push

**Comment by milo** [comment]
@lukyer :+1: So the last thing is the commit message to be consistent :) Something like `Session: session ID is not regenerated when not set`

**Comment by lukyer** [comment]
@milo Thanks for your tutorial :) I must admit, pretty complicated way to change few letters in Nette code :) Hope it will get better with better git cli skills.

Btw: I even wasn't able to do this simple operation (squash last 2 commits) in Intellij PhpStorm :(

File: src/Http/Session.php [hunk 2]
```diff
@@ -226,7 +226,9 @@ public function regenerateId()
 			if (headers_sent($file, $line)) {
 				throw new Nette\InvalidStateException('Cannot regenerate session ID after HTTP headers have been sent' . ($file ? " (output started at $file:$line)." : '.'));
 			}
-			session_regenerate_id(TRUE);
+			if (session_id() !== '') {
+				session_regenerate_id(TRUE);
+			}
 			session_write_close();
```
Comment (dg) [inline src/Http/Session.php]: session_write_close works when `session_id() !== ''` ?

Comment (lukyer) [inline src/Http/Session.php]: It seems working, but this situation is kinda rare, so if another error will show up, i will fix it here.

Comment (dg) [inline src/Http/Session.php]: Why it is rare?

Comment (lukyer) [inline src/Http/Session.php]: Probably nobody knows. Maybe PHP bug. I suspect altered session_name.
See https://github.com/yiisoft/yii/issues/2376

Comment (dg) [inline src/Http/Session.php]: When it is empty on line 229, it should be empty on next line 232 too, isn't it?

**Comment by dg** [comment]
Do you use default or custom session handler? In case of default file handle, what is option session.save_path value?

**Comment by lukyer** [comment]
Default one.
session.save_path = 'C:\xampp\tmp'
session.name = 'FDV'

**Pull request merged**